sdcm/tester.py

```diff
@@ -905,10 +941,30 @@ def run_fullscan_thread(self, ks_cf='random', interval=1, duration=None):
         """
         duration = self.get_duration(duration)
         interval = interval * 60
-        self.loaders.run_fullscan_thread(ks_cf,
-                                         db_nodes=self.db_cluster.nodes,
-                                         interval=interval,
-                                         duration=duration)
+
+        @log_run_info('Fullscan thread')
+        def run_in_thread():
+            start = current = time.time()
+            while current - start < duration:
+                loader_node = random.choice(self.loaders.nodes)
+                db_node = random.choice(self.db_cluster.nodes)
+                if db_node.termination_event.isSet():
```

Comment:
we can improve and move this in the while condition check

Reply:
@bentsi  could you explain please?

Reply:
I want that every time to run the full scan from another node, not from same

Reply:
ok, looking on it again, this is an issue, Consider
We don't need to run on db_node when terminate and replace/decommission nemesis run 
and need ability to terminate the thread at the end of the test